dds/DCPS/XTypes/DynamicTypeSupport.cpp

```diff
@@ -208,6 +208,31 @@ namespace OpenDDS {
 
   }
 
+  namespace DCPS {
+#ifndef OPENDDS_NO_CONTENT_SUBSCRIPTION_PROFILE
+    template <>
+    DDS::ReturnCode_t DataReaderImpl_T<XTypes::DynamicSample>::read_generic(
+      GenericBundle&,
+      DDS::SampleStateMask,
+      DDS::ViewStateMask,
+      DDS::InstanceStateMask,
+      bool)
+    {
+      return DDS::RETCODE_UNSUPPORTED;
+    }
```

Comment:
@mitza-oci I moved this and `DataReaderImpl_T<XTypes::DynamicSample>::take` here from `DynamicDataReaderImpl.cpp` after getting this very unhelpful link error while fixing a different issue with `scrap_vs.pl` in the static CMake Windows CI build (`build_cmake_w22_stat`):

```
OpenDDS_Dcps.lib(unity_13_cxx.obj) : error LNK2019: unresolved external symbol "public: virtual int __cdecl OpenDDS::DCPS::DataReade
rImpl_T<class OpenDDS::XTypes::DynamicSample>::read_generic(struct OpenDDS::DCPS::DataReaderImpl::GenericBundle &,unsigned int,unsig
ned int,unsigned int,bool)" (?read_generic@?$DataReaderImpl_T@VDynamicSample@XTypes@OpenDDS@@@DCPS@OpenDDS@@UEAAHAEAUGenericBundle@D
ataReaderImpl@23@III_N@Z) referenced in function "[thunk]:public: virtual int __cdecl OpenDDS::DCPS::DataReaderImpl_T<class OpenDDS:
:XTypes::DynamicSample>::read_generic`vtordisp{4294967292,8}' (struct OpenDDS::DCPS::DataReaderImpl::GenericBundle &,unsigned int,un
signed int,unsigned int,bool)" (?read_generic@?$DataReaderImpl_T@VDynamicSample@XTypes@OpenDDS@@@DCPS@OpenDDS@@$4PPPPPPPM@7EAAHAEAUG
enericBundle@DataReaderImpl@23@III_N@Z) [C:\Users\Fred\oci\w19_re_o1p1_sec\OpenDDS\build\DevGuideExamples\DCPS\Messenger\subscriber.
vcxproj]
OpenDDS_Dcps.lib(unity_13_cxx.obj) : error LNK2019: unresolved external symbol "public: virtual int __cdecl OpenDDS::DCPS::DataReade
rImpl_T<class OpenDDS::XTypes::DynamicSample>::take(class OpenDDS::DCPS::AbstractSamples &,unsigned int,unsigned int,unsigned int)"
(?take@?$DataReaderImpl_T@VDynamicSample@XTypes@OpenDDS@@@DCPS@OpenDDS@@UEAAHAEAVAbstractSamples@23@III@Z) referenced in function "[
thunk]:public: virtual int __cdecl OpenDDS::DCPS::DataReaderImpl_T<class OpenDDS::XTypes::DynamicSample>::take`vtordisp{4294967292,8
}' (class OpenDDS::DCPS::AbstractSamples &,unsigned int,unsigned int,unsigned int)" (?take@?$DataReaderImpl_T@VDynamicSample@XTypes@
OpenDDS@@@DCPS@OpenDDS@@$4PPPPPPPM@7EAAHAEAVAbstractSamples@23@III@Z) [C:\Users\Fred\oci\w19_re_o1p1_sec\OpenDDS\build\DevGuideExamp
les\DCPS\Messenger\subscriber.vcxproj]
C:\Users\Fred\oci\w19_re_o1p1_sec\OpenDDS\build\DevGuideExamples\DCPS\Messenger\Debug\subscriber.exe : fatal error LNK1120: 2 unreso
lved externals [C:\Users\Fred\oci\w19_re_o1p1_sec\OpenDDS\build\DevGuideExamples\DCPS\Messenger\subscriber.vcxproj]
```

This was mostly a shot in the dark, but because this was a unity build I looked and saw `DynamicDataReaderImpl.cpp` was in `unity_12_cxx.cxx` and not `unity_13_cxx` mentioned in the error. In `unity_13_cxx.cxx`, I think `DynamicTypeSupport.cpp` is the only file that uses `DynamicDataReader` so I moved the functions here and that solved the issue. I'm not entirely sure I understand the problem through because normally in a non-unity build they are not in the same translation unit, so this shouldn't matter as far as I know. If you have any alternative suggestions I'd be happy to try them out.

Reply:
I'm not sure, what do I need to do to enable unity build?
In general I'd rather turn unity build off for now

Reply:
#4308